packages/cli/src/cmds/stats/stats.ts

```diff
@@ -0,0 +1,319 @@
+
+      // This function is too verbose to be useful in this context.
+      const v = verbose();
+      verbose(false);
+      await listAppMapFiles(appMapDir, (fileName: string) => {
+        const stats = fs.statSync(fileName);
+        appMapSizes[fileName] = {
+          size: stats.size,
+        };
+      });
+      verbose(v);
+
+      return appMapSizes;
+    }
+
+    async function sortAppMapSizes(appMapSizes) {
+      let appMapSizesArray: { name: string; size: number }[] = [];
+
+      for (const key in appMapSizes) {
+        appMapSizesArray.push({
+          name: key,
+          size: appMapSizes[key].size,
+        });
+      }
+
+      return appMapSizesArray.sort((a, b) => b.size - a.size);
+    }
+
+    function sizeInMB(size) {
+      return Number((size / 1000 / 1000).toFixed(1));
+    }
+
+    async function calculateExecutionTimes(appMapDir) {
+      let functionExecutionTimes = {};
+
+      // This function is too verbose to be useful in this context.
+      const v = verbose();
+      verbose(false);
+      await listAppMapFiles(appMapDir, (fileName: string) => {
+        const file = fs.readFileSync(fileName, 'utf-8');
+        let appmap = JSON.parse(file.toString());
+        functionExecutionTimes[fileName] = {
+          functions: {},
+        };
+
+        appmap.events.forEach((event) => {
+          if (event.event == 'call') {
+            const functionKey = 'thread_' + event.thread_id + '_id_' + event.id;
+            if (!(functionKey in functionExecutionTimes[fileName].functions)) {
+              let name = '';
+              if ('defined_class' in event) {
+                name = event.defined_class + '.' + event.method_id;
+              } else if ('sql_query' in event) {
+                name = event.sql_query;
+              } else if ('http_server_request' in event) {
+                name =
+                  event.http_server_request.request_method +
+                  ' ' +
+                  event.http_server_request.path_info;
+              }
+              let newData = {
+                name: name,
+                elapsedInstrumentationTimeTotal: 0,
+                callNoReturnFound: true,
+                numberOfCalls: 0,
+                path: '',
+              };
+              if ('defined_class' in event && event.path[0] == '/') {
+                // Absolute path names generally signify a library function.
+                // Send library function data to help optimize AppMap;
+                // don't send user function data.
+                newData.path = event.path;
+              }
+              functionExecutionTimes[fileName].functions[functionKey] = newData;
+            } else {
+              functionExecutionTimes[fileName].functions[
+                functionKey
+              ].callNoReturnFound = true;
+            }
+          } else if (event.event == 'return') {
+            // A return's parent_id matches the call's id
+            const functionKey =
```

Comment:
You can get the call and return events from any event using event.calEvent and event.returnEvent